syntax = "proto3";
package eos.ns;

//------------------------------------------------------------------------------
// Container metadata protocol buffer object
//------------------------------------------------------------------------------
message ContainerMdProto {
  uint64 id = 1;
  uint64 parent_id = 2;
  int64 uid = 3;
  int64 gid = 4;
  int64 tree_size = 6;
  int32 mode = 5;
  uint32 flags = 7;
  string name = 8;
  bytes ctime = 9; // change time
  bytes mtime = 10; // modification time
  bytes stime = 11; // sync time
  map<string, string> xattrs = 12;
  // @todo (esindril): This is not actually used, we should drop it
  uint32 acl_id = 13;
}
